### PR TITLE
Proxy precompile in Frontier container chain runtime

### PR DESCRIPTION
**Pull request by fgamundi**
Add Proxy precompiled smart contract from Moonkit to Frontier container chain at address `0x0000000000000000000000000000000000000805`

### CODE REVIEW

**Review by girazoki — APPROVED** [review]
LGTM

**Comment by girazoki** [comment]
@fgamundi merge master back in please

File: container-chains/runtime-templates/frontier/src/migrations.rs [hunk 1]
```diff
@@ -77,6 +77,53 @@ where
     }
 }
 
+pub struct MigratePrecompileProxyDummyCode<T>(pub PhantomData<T>);
```
Comment (tmpolaczyk) [inline container-chains/runtime-templates/frontier/src/migrations.rs:80]: Maybe in the future we can have a generic migration in moonkit that we pass it the precompile address and it automatically writes the bytecode?

Comment (fgamundi) [reply container-chains/runtime-templates/frontier/src/migrations.rs:80]: That's a neat idea. For sure let's do that for the next one

**Review by Agusrodri — APPROVED** [review]

**Pull request merged**